### PR TITLE
config: add PBX related options

### PR DESCRIPTION
**Pull request by Amygos**
Add pbx.provider and pbx.provider.nethvoice.webrest.url
NethServer/dev#5516

### CODE REVIEW

File: root/etc/e-smith/events/actions/nethserver-webtop5-conf-db [hunk 1]
```diff
@@ -118,6 +118,20 @@ my $davserver_url = $db->get_prop('webtop','DavServerUrl') || 'http://'.$db->get
...
 
+# Set pbx provider
+my $pbx_provider = $db->get_prop('webtop','PbxProvider') || "";
+
+print $fh "DELETE FROM \"core\".\"settings\" WHERE service_id = 'com.sonicle.webtop.core' and key = 'pbx.provider';\n";
+print $fh "INSERT INTO \"core\".\"settings\" (\"service_id\", \"key\", \"value\") VALUES ('com.sonicle.webtop.core', 'pbx.provider', '$pbx_provider');\n";
+
+if ($pbx_provider eq "nethvoice") {
+    # Set NethVoice base url for api calls
+    my $pbx_provider_nethvoice_webrest_url = $db->get_prop('webtop','PbxProviderNethvoiceWebrestUrl') || "";
```
Comment (gsanchietti) [inline root/etc/e-smith/events/actions/nethserver-webtop5-conf-db]: I don't really like to hard code the "NethVoice" label inside the property name, but I feel is the best option to map the prop to the database value.

I'm wondering why WebTop developers need to specify the provider name twice: once as value of  `pbx.provider` option and again inside `pbx.provider.nethvoice.webrest.url` option name.

The only reason I see, is if multiple providers can be used at the same time. But I don't even see a usage scenario: I guess is quite unlikely that after a click a call is started on two different PBX.

To sum up, I propose an upstream change:
rename `pbx.provider.nethvoice.webrest.url` to `pbx.provider.webrest.url` *or* just remove the `pbx.provider`option.


@lucagasparini @gbulfon, what do you think?

Comment (gbulfon) [reply root/etc/e-smith/events/actions/nethserver-webtop5-conf-db]: Yes, the reason is to possibly support different providers in the future, so the first setting is there to tell the provider code-name.
The separate setting has the code-name in the property name because every provider may have its own ones, with different meanings. Now it's the webrest url, tomorrow it may be something very specific to NethVoice.
I don't see the issue with having it there, unless you're worried about a possible product name change.

Comment (gsanchietti) [reply root/etc/e-smith/events/actions/nethserver-webtop5-conf-db]: "nethvoice" is a commercial product name and shouldn't be hard-coded anywhere.

By the was, I'd like to not waste more time on it.

@Amygos, feel free to merge.
